modules/indigo/module/inc/indigo/of_connection_manager.h

```diff
@@ -814,6 +814,49 @@ indigo_cxn_subbundle_set2(uint32_t num_subbundles,
                           indigo_cxn_subbundle_start_t starts[],
                           indigo_cxn_subbundle_finish_t finishes[]);
 
+
+/**
+ * subbundle functions are used to communicate with driver that the
+ * bundle task is going to start the subbundle process and bundle task
+ * has processed the last message in the subbundle.
+ * Driver can take the proper action accordingly.
+ */
+typedef struct indigo_cxn_subbundle_info_s {
+    uint32_t subbundle_idx;
+    uint32_t total_msg_count;
+    uint32_t cur_msg_count;
+} indigo_cxn_subbundle_info_t;
+
+typedef void (*indigo_cxn_subbundle_pre_start_t)(indigo_cxn_id_t cxn_id,
+                                                 indigo_cxn_subbundle_info_t *subbundle_info);
+typedef void (*indigo_cxn_subbundle_post_finish_t)(indigo_cxn_id_t cxn_id,
+                                                   indigo_cxn_subbundle_info_t *subbundle_info);
+
+/**
+ * Extends the subbundle infrastructure from indigo_cxn_subbundle_set2 above.
+ * First five parameters are the same as for indigo_cxn_subbundle_set2.
+ * @param pre_starts Array of prestart functions; the ith function is called
+ * before the ith subbundle is started.
+ * @param post_finishes Array of postfinish functions; the ith function is called
+ * after the ith subbundle is finished.
+ *
+ * The intent is to provide subbundle information to the the flowtables or
+ * gentables before or after processing the subbundle.
+ * Adding these two functions also try to backward compatible the existing
+ * codes instead of modifying the start() and finish() functions.
```

Comment:
let's just delete lines 845-846, and replace with "call only one of indigo_cxn_subbundle_set, indigo_cxn_subbundle_set2, or indigo_cxn_subbundle_set3 functions"